assets/src/components/form/model/base/model-select.js

```diff
@@ -130,6 +159,71 @@ export class ModelSelect extends Component {
 		return null;
 	}
 
+	/**
+	 * Helper method for determining whether state should be updated.
+	 * Criteria is whether the keys from the incoming entities are equivalent
+	 * (equivalency means in same order as well)
+	 *
+	 * @param {Object} prevProps
+	 * @param {Object} nextProps
+	 * @return {boolean} True means state should be updated.
+	 */
+	static shouldUpdateState( prevProps, nextProps ) {
+		if ( prevProps === nextProps ) {
+			return false;
+		}
+
+		const { selectConfiguration: prevConfiguration } = prevProps;
+		const { selectConfiguration: nextConfiguration } = nextProps;
+
+		// if defaultValue has changed (selected value) then update state
+		if (
+			prevConfiguration.defaultValue !== nextConfiguration.defaultValue
+		) {
+			return true;
+		}
+
+		if ( prevConfiguration.isLoading !== nextConfiguration.isLoading ) {
+			return true;
+		}
+
+		// shallow compare of keys but only if we can
+		if (
+			! isMap( prevProps.modelEntities ) ||
+			! isMap( nextProps.modelEntities )
+		) {
+			return true;
+		}
```

Comment:
you are only highlighting a portion of the comment.  In it's entirety the comment is "shallow compare of keys *but only if we can*" (extra emphasis added) - so the intent being to communicate that the shallow compare of keys only happens if the conditional below passes.  

> those two modelEntities could both be Map objects but have completely different keys and this block would get skipped, no?

So if this block (the conditional checks for Maps) is skipped, then that means we _can_ do a shallow compare of keys (the next block).  The is Map conditional check is done (and can trigger a re-render) because its possible on initialization that there are no Maps for either props being compared.